executor.py

```diff
@@ -74,8 +81,25 @@ def search(
             match_args.update(parameters)
 
         match_args = SimpleIndexer._filter_parameters(docs, match_args)
-
-        docs.match(self._storage, **match_args)
+        left_trav_path = match_args.get('traversal_ldarray')
+        # if it's 'r' we would just be duplicating
+        if left_trav_path and left_trav_path != 'r':
+            left_trav_docs = DocumentArray(docs.traverse_flat(left_trav_path))
```

Comment:
I think you should also pass the `filter_fn` args to `traverse_flat()` and remove it from the `match_args` since it will be [deprecated too](https://github.com/jina-ai/jina/blob/master/jina/types/arrays/mixins/traverse.py#L122).

Reply:
When it gets deprecated. Not in this PR